go-packages/meep-model/model.go

```diff
@@ -957,590 +956,211 @@ func (m *Model) GetNodeContext(name string) (ctx interface{}) {
-			} else {
-				nList = m.nodeMap.FindAllByType(NodeTypeOperatorCell)
-				nListOther := m.nodeMap.FindAllByType(NodeTypeOperator)
-				nList = append(nList, nListOther...)
-			}
-		}
-	case Zone:
-		if filter.ZoneName != "" {
-			n = m.nodeMap.FindByName(filter.ZoneName)
-			if n != nil {
-				nList = append(nList, n)
-			}
-		} else {
-			nList = m.nodeMap.FindAllByType(NodeTypeZone)
-		}
-		//only one type of zone... so no need to check for a different filter
-	case NetLoc:
-		if filter.NetworkLocationName != "" {
-			if filter.NetworkLocationType != "" {
-				n = m.nodeMap.FindByType(filter.NetworkLocationName, filter.NetworkLocationType)
-				if n != nil {
-					nList = append(nList, n)
-				}
-			} else {
-				n = m.nodeMap.FindByName(filter.NetworkLocationName)
-				if n != nil {
-					nList = append(nList, n)
-				}
-			}
-		} else {
-			if filter.NetworkLocationType != "" {
-				nList = m.nodeMap.FindAllByType(filter.NetworkLocationType)
-			} else {
-				nList = m.nodeMap.FindAllByType(NodeTypePoa)
-				nListOther := m.nodeMap.FindAllByType(NodeTypePoa4G)
-				nList = append(nList, nListOther...)
-				nListOther = m.nodeMap.FindAllByType(NodeTypePoa5G)
-				nList = append(nList, nListOther...)
-				nListOther = m.nodeMap.FindAllByType(NodeTypePoaWifi)
-				nList = append(nList, nListOther...)
-			}
-		}
-	case PhyLoc:
-		if filter.PhysicalLocationName != "" {
-			if filter.PhysicalLocationType != "" {
-				n = m.nodeMap.FindByType(filter.PhysicalLocationName, filter.PhysicalLocationType)
-				if n != nil {
-					nList = append(nList, n)
-				}
-			} else {
-				n = m.nodeMap.FindByName(filter.PhysicalLocationName)
-				if n != nil {
-					nList = append(nList, n)
+	// Find nodes that match filter criteria
+	var domains dataModel.Domains
+	domains.Domains = []dataModel.Domain{}
+	for _, node := range nMap {
+		if m.filterNode(node, Domain, filter) {
+			var domain dataModel.Domain
+
+			// Deep copy node if it needs to be modified
+			obj := node.object.(*dataModel.Domain)
+			if !filter.Children || filter.Minimize {
+				byt, _ := json.Marshal(obj)
+				err := json.Unmarshal(byt, &domain)
+				if err != nil {
+					continue
 				}
-			}
-		} else {
-			if filter.PhysicalLocationType != "" {
-				nList = m.nodeMap.FindAllByType(filter.PhysicalLocationType)
-			} else {
-				nList = m.nodeMap.FindAllByType(NodeTypeCloud)
-				nListOther := m.nodeMap.FindAllByType(NodeTypeEdge)
-				nList = append(nList, nListOther...)
-				nListOther = m.nodeMap.FindAllByType(NodeTypeFog)
-				nList = append(nList, nListOther...)
-				nListOther = m.nodeMap.FindAllByType(NodeTypeUE)
-				nList = append(nList, nListOther...)
-			}
-		}
-	case Proc:
-		if filter.ProcessName != "" {
-			if filter.ProcessType != "" {
-				n = m.nodeMap.FindByType(filter.ProcessName, filter.ProcessType)
-				if n != nil {
-					nList = append(nList, n)
+
+				// Remove children
+				if !filter.Children {
+					domain.Zones = nil
```

Comment:
The logic `!filter.Children` is awkward
- if `filter.children == false` then childrens are removed
- if `filter.children == true` then childrens are kept

Reply:
Agreed. I will update API & backend accordingly.

Reply:
NOTE: I will push the necessary updates in an upcoming PR.